odoo/addons/base/tests/test_ir_http.py

```diff
@@ -28,3 +30,16 @@ def test_routing_map_performance(self):
         self.env['ir.http'].routing_map(key=1)
         duration = time.time() - start
         _logger.info('Routing map website1 generated in %.3fs', duration)
+
+class TestIrHttp(TransactionCase):
+    @patch("odoo.http.root.session_store.vacuum")
+    def test_gc_session_ignored_no_db_name(self, mock):
+        with patch.object(config, 'options', {**config.options, 'db_name': get_db_name()}):
+            self.env['ir.http']._gc_sessions()
+            mock.assert_called_once()
+
+        # Make like we didn't provide the -d / --database option
+        with patch.object(config, 'options', {**config.options, 'db_name': False}):
+            self.env['ir.http']._gc_sessions()
+            mock.reset_mock()
```

Comment:
the `reset_mock` isn't at the right place, it should be before `_gc_sessions` at least